src/manifolds/Stiefel.jl

```diff
@@ -287,6 +373,20 @@ where $\operatorname{sgn}(p) = \begin{cases}
 """
 retract(::Stiefel, ::Any, ::Any, ::QRRetraction)
 
+function retract!(::Stiefel, q, p, X, ::PadeRetraction{m}) where {m}
+    Pp = p * p'
+    Pp .= one(Pp) - Pp
```

Comment:
```suggestion
    Pp = I - p * p'
```
This should be faster.

Reply:
Thanks, I also fixed the second occurence.